engine/gamesys/src/gamesys/test/resource/set_texture.script

```diff
@@ -77,21 +77,45 @@ local function test_fail_wrong_mipmap(self)
 		height       = height,
 		type         = resource.TEXTURE_TYPE_2D,
 		format       = resource.TEXTURE_FORMAT_RGB,
-		mipmap       = 128,
+		mipmap       = 31,
 	}
 	resource.set_texture(resource_path, header, buf)
 end
 
-function update(self, dt)
-	if self.update_counter == 0 then
-		test_success_simple(self)
-	elseif self.update_counter == 1 then
-		test_success_resize(self)
-	elseif self.update_counter == 2 then
-		test_fail_out_of_bounds(self)
-	elseif self.update_counter == 2 then
-		test_fail_wrong_mipmap(self)
+local function test_success_compressed(self)
+	local f     = io.open("src/gamesys/test/resource/blank.basis", "rb")
```

Comment:
This won't work on Switch/PS4/PS5 tests (or Android/iOS later on for that matter).

You should use the `DM_HOSTFS` define that the build system sets, and pass that to the Lua script somehow.